### ISSUE TITLE
Major-ordering of 'delete', select-by-alias

### ISSUE THREAD

**Issue by maxvu** [elastic/curator#571]
As a kind of logical structuring, my organization prefixes each series name with an identifier for the owning project/group (e.g. 'common-apache-2016.02.23' for the 'apache' series, which belongs to group 'common'). When we want to select with curator a specific series, we're careful to name prefixes up to and including that dash character separating the date. We've recently switched our logging over from a nearly identical naming scheme, excepting the group/series separator being an underbar ('_') character instead of a dash character.

We are fine naming operations twice (once per series, per scheme), but we're running into trouble with our recurring maintenance script that invokes 'delete': we're finding that this change violates the verb's expectation of indices' major-, lexicographic-ordering quality. For 'delete' with '--disk-space' defined, the order in which the indices are considered is crucial to operation: it determines the order of accumulation and, eventually, which indices will be deleted.

Consider the following sample that, I think, demonstrates the [part of the code](https://github.com/elastic/curator/blob/v3.2.2/curator/cli/index_selection.py#L109) that we're interested in:

```
	>>> sorted( [
		'common_apache-2000.01.02',
		'common_apache-2000.01.03',
		'common_apache-2000.01.04',
		'common-apache-2000.01.05v1',
		'common-apache-2000.01.06v1'
	] )
	
	[
		'common-apache-2000.01.05v1',
		'common-apache-2000.01.06v1',
		'common_apache-2000.01.02',
		'common_apache-2000.01.03',
		'common_apache-2000.01.04'
	]
```

Here, we want to treat both indices separated by an underbar and indices separated by a dash as being part of the same series: when the newest dashed names grow to a certain size, begin deleting the oldest underbar names. However, the example (together with the code linked) should indicate that, while we can select the right subset of indices, we cannot have them considered in the correct order.

Certainly, we're responsible for our own naming and I wouldn't be surprised to learn that this is a rare case, but there are two specific details that have me considering how difficult it would be to accomodate within the script:

1. We have fully aliased, one-to-one, each old-scheme name to its corresponding new-scheme name, and in both directions. With those aliases, Kibana seems to respect this mapping and can navigate the indices between schemes without issue. If the selection criteria pertaining to names could be expanded to match an index with an alias of the same shape, then we could select for just one scheme, be returned all indices we're concerned about and their names would have the above-named ordering quality.

2. We do specify the timestring for each invocation. It looks as though that that option is currently used only to match out index names that have the right number and placement of numerals, but it should also be possible to infer from it the time unit it corresponds to and order results by that instead.

I'm not too familiar with the specifics of many-to-many aliasing in Elasticsearch and haven't spent too much time thinking about how other verbs would be affected, but thought that I should at least articulate the problem and see how the maintainers feel about it.

**Comment by maxvu** [comment]
I notice now that there is an open pull request ([#517](https://github.com/elastic/curator/pull/517#issuecomment-161459718)) that looks to solve a similar problem.

**Comment by maxvu** [comment]
After studying the Elasticsearch documentation a little closer, it seems as though [aliases are endorsed pretty strongly](https://www.elastic.co/guide/en/elasticsearch/guide/current/index-aliases.html) ("Aliases are cheap and should be used liberally"). Although I feel both proposals given would solve the problem given, I think a solution that involves accommodating aliases would be more useful on-the-whole.

**Comment by untergeek** [comment]
Ordering by age with delete by space is addressed by #595 and #596 with Curator 4.0 imminent.

Please open a separate feature request for select by alias.
